components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionSupport.kt

```diff
@@ -140,19 +144,25 @@ object WebExtensionSupport {
                 engineSession: EngineSession,
                 action: Action
             ): EngineSession? {
-                val popupSessionId = store.state.extensions[webExtension.id]?.browserActionPopupSession
-                return if (popupSessionId != null && store.state.tabs.find { it.id == popupSessionId } != null) {
-                    if (popupSessionId == store.state.selectedTabId) {
-                        closeTab(popupSessionId, store, onCloseTabOverride, webExtension)
+                return if (onActionPopupToggleOverride != null) {
+                    onActionPopupToggleOverride.invoke(webExtension)
+                    store.dispatch(WebExtensionAction.UpdatePopupSessionAction(webExtension.id, popupSession = engineSession))
```

Comment:
nit: I think we should first dispatch the action, then invoke the override, so the store is updated...it should work either way though.